src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -881,6 +893,20 @@ private static Conversion ClassifyNullLiteralConversion(BoundExpression source,
             return Conversion.NoConversion;
         }
 
+        private Conversion ClassifyDefaultConversion(BoundDefaultOperator source, TypeSymbol destination, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        {
+            Debug.Assert((object)source != null);
+            Debug.Assert((object)source.Type == null);
+            Debug.Assert((object)destination != null);
+
+            if (destination.IsStructType() && !destination.IsNullableType())
+            {
+                return Conversion.DefaultLiteral;
+            }
+
+            return Conversion.NoConversion;
```

Comment:
The idea was that you should use `null` for ref types and `default` for struct types. But there is also an argument that `default` should be allowed anywhere `default(type)` was previous allowed.
I'll remove